File: LeetCode/199.php
<?php


class TreeNode
{
    public $val = null;
    public $left = null;
    public $right = null;
    function __construct($value)
    {
        $this->val = $value;
    }
}

class Solution
{

    /**
     * @param TreeNode $root
     * @return Integer[]
     */
    function rightSideView($root)
    {
        $queue =  $result = [];
        $queue[] = $root;
        while (!empty($queue)) {
            $result[] = $queue[0]->val;
            $temp = [];
            foreach ($queue as $node) {
                if ($node->right)
                    $temp[] = $node->right;
                if ($node->left)
                    $temp[] = $node->left;
            }
            $queue = $temp;
        }
        return $result;
    }
}

$root = new TreeNode(1);
$root->left = new TreeNode(2);
$root->left->right = new TreeNode(5);
$root->right = new TreeNode(3);
// $root->right->right = new TreeNode(4);
$test = new Solution();
var_dump($test->rightSideView($root));

File: LeetCode/199.java
/**
 * Definition for a binary tree node. public class TreeNode { int val; TreeNode
 * left; TreeNode right; TreeNode(int x) { val = x; } }
 */

class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        if (root != null) {
            Queue<TreeNode> queue = new LinkedList<>();
            queue.offer(root);
            while (queue.size() != 0) {
                result.add(queue.peek().val);
                int size = queue.size();
                while (size-- != 0) {
                    TreeNode temp = queue.poll();
                    if (temp.right != null)
                        queue.offer(temp.right);
                    if (temp.left != null)
                        queue.offer(temp.left);
                }
            }
        }
        return result;
    }
}